catroid/src/androidTestPhiro/java/org/catrobat/catroid/uiespresso/ui/activity/PhiroScriptActivityTest.java

```diff
@@ -0,0 +1,126 @@
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.bricks.SetXBrick;
+import org.catrobat.catroid.ui.ScriptActivity;
+import org.catrobat.catroid.uiespresso.util.BaseActivityInstrumentationRule;
+import org.catrobat.catroid.uiespresso.util.UiTestUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_SHOW_HINTS;
+import static org.catrobat.catroid.ui.SettingsActivity.SETTINGS_SHOW_PHIRO_BRICKS;
+import static org.catrobat.catroid.uiespresso.ui.activity.utils.FormulaEditorCategoryListDataInteractionWrapper.onFormulaEditorCategoryItemWithNameAtPosition;
+import static org.catrobat.catroid.uiespresso.ui.activity.utils.ScriptListDataInteractionWrapper.onBrickCategoryWithNameAtPosition;
+
+@RunWith(AndroidJUnit4.class)
+public class PhiroScriptActivityTest {
+
+	private Map<String, Boolean> initialPreferences = new HashMap<>();
+
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		createProject("phiroTestProject");
+		preparePhiroSettings();
+
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@After
+	public void teardown() throws Exception {
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+		SharedPreferences.Editor editor = sharedPreferences.edit();
+
+		editor.putBoolean(SETTINGS_SHOW_PHIRO_BRICKS, initialPreferences.get(SETTINGS_SHOW_PHIRO_BRICKS));
+		editor.putBoolean(SETTINGS_SHOW_HINTS, initialPreferences.get(SETTINGS_SHOW_HINTS));
+		editor.apply();
+	}
+
+	@Test
+	public void phiroIsFirstInBrickCategoryTest() {
+		onView(withId(R.id.button_add))
+				.perform(click());
+
+		onBrickCategoryWithNameAtPosition("Phiro", 0)
+				.check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void phiroIsFirstInFormulaEditorDeviceCategoryTest() {
+		onView(withId(R.id.brick_set_x_edit_text))
+				.perform(click());
+
+		onView(withId(R.id.formula_editor_keyboard_sensors))
+				.perform(click());
+
+		String firstPhiroSensorString = UiTestUtils.getResourcesString(R.string.formula_editor_phiro_sensor_front_left);
+
+		onFormulaEditorCategoryItemWithNameAtPosition(firstPhiroSensorString, 0)
```

Comment:
OK, this is possible now for both, FE and Brick categories:
onFormulaEditorCategory()
				.atPosition(0)
				.check(matches(withText(firstPhiroSensorString)));
..and:
onBrickCategory()
				.atPosition(0)
				.check(matches(withText(expectedCategoryString)));

I've left them in wrapper functions since I had to check for the childView(..) of the respective LinearLayout, too. I think therefore it's worth to put it inside a wrapper.